rdr_service/offline/retention_eligible_import.py

```diff
@@ -43,36 +46,62 @@ def import_retention_eligible_metrics_file(task_data):
 
     batch_count = upsert_count = 0
     records = list()
+    failed_records_count = 0
     with dao.session() as session:
         for row in csv_reader:
-            if not row[RetentionEligibleMetricCsvColumns.PARTICIPANT_ID]:
-                continue
-            record = _create_retention_eligible_metrics_obj_from_row(row, upload_date)
-
-            existing_id, needs_update = RetentionEligibleMetricsDao.find_metric_with_session(
-                session=session,
-                metrics_obj=record
-            )
-            if existing_id:
-                record.id = existing_id
-            if needs_update:
-                _supplement_with_rdr_calculations(metrics_data=record, session=session)
-                records.append(record)
-                batch_count += 1
-
-            if batch_count == _BATCH_SIZE:
-                upsert_count += dao.upsert_all_with_session(session, records)
+            try:
+                if not row[RetentionEligibleMetricCsvColumns.PARTICIPANT_ID]:
+                    continue
+                record = _create_retention_eligible_metrics_obj_from_row(row, upload_date)
+
+                existing_id, needs_update = RetentionEligibleMetricsDao.find_metric_with_session(
+                    session=session,
+                    metrics_obj=record
+                )
+                if existing_id:
+                    record.id = existing_id
+                if needs_update:
+                    _supplement_with_rdr_calculations(metrics_data=record, session=session)
+                    records.append(record)
+                    batch_count += 1
+
+                if batch_count == _BATCH_SIZE:
+                    upsert_count += dao.upsert_all_with_session(session, records)
+                    records.clear()
+                    batch_count = 0
+            except (IntegrityError, InvalidRequestError):
+                failed_records_count += batch_count
                 records.clear()
                 batch_count = 0
+                continue
 
         if records:
-            upsert_count += dao.upsert_all_with_session(session, records)
+            try:
+                upsert_count += dao.upsert_all_with_session(session, records)
+            except (IntegrityError, InvalidRequestError):
+                failed_records_count += batch_count
+
+        _send_slack_alert(f'gs://{csv_file_cloud_path}', failed_records_count)
 
     logging.info(f"Updating participant summary retention eligible flags for {upsert_count} participants...")
     ParticipantSummaryDao().bulk_update_retention_eligible_flags(upload_date)
     logging.info(f"Import and update completed for gs://{csv_file_cloud_path}")
 
 
+def _send_slack_alert(file_name, failed_records):
```

Comment:
Since this isnt part of a class dont think this needs to be denoted as a protected method.